File: Server/API/models/event.js
const mongoose = require('mongoose');

const eventSchema= mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    name: {type:String, required:true},
    contact: {type: String, required: false},
    start: {type: Date, required: true},
    end: {type: Date, required: true},
    room: {type: String, required: true},
    type: {type: String, required: true},
    department: {type: String, reqired: true},
    website: {type: String, reqired: false},
    description: {type: String, reqired: false},
    image: {type: String, default: 'uploads\\haw.png'}
});

module.exports = mongoose.model('Event',eventSchema);

File: haw_events_application/lib/event.dart
class EventList {
  final int count;
  final List<Event> events;

  EventList({
    this.count,
    this.events,
  });

  factory EventList.fromJson(Map<String, dynamic> parsedJson) {
    var events = parsedJson['events'] as List;
    print(events.runtimeType);
    List<Event> eventList = events.map((i) => Event.fromJson(i)).toList();

    return EventList(
      count: parsedJson['count'],
      events: eventList
    );
  }
}

class Event {

  String image;
  String id;
  String name;
  String contact;
  String start;
  String end;
  String room;
  String type;
  String department;
  String website;
  String description;

  Event ({
    this.image,
    this.id,
    this.name,
    this.contact,
    this.start,
    this.end,
    this.room,
    this.type,
    this.department,
    this.website,
    this.description,
  });

  factory Event.fromJson(Map<String, dynamic> parsedJson){
    return Event(
    image: parsedJson['image'],
    id: parsedJson['_id'],
    name: parsedJson['name'],
    contact: parsedJson['contact'],
    start: parsedJson['start'],
    end: parsedJson['end'],
    room: parsedJson['room'],
    type: parsedJson['type'],
    department: parsedJson['department'],
    website: parsedJson['website'],
    description: parsedJson['description'],
    );
  }
}